File: ios-mma-recomp/MMARecomp/Models/NutritionTarget.swift
import Foundation

struct NutritionTarget: Codable, Identifiable {
    var id: UUID
    var userId: UUID
    var date: String
    var typeJour: TypeJour
    var caloriesCible: Int
    var proteinesCibleG: Double
    var createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case date
        case typeJour = "type_jour"
        case caloriesCible = "calories_cible"
        case proteinesCibleG = "proteines_cible_g"
        case createdAt = "created_at"
    }
}

struct NewNutritionTarget: Codable {
    var date: String
    var typeJour: TypeJour
    var caloriesCible: Int
    var proteinesCibleG: Double

    enum CodingKeys: String, CodingKey {
        case date
        case typeJour = "type_jour"
        case caloriesCible = "calories_cible"
        case proteinesCibleG = "proteines_cible_g"
    }
}

File: android-mma-recomp/app/src/main/java/com/example/mmarecomp/model/NutritionTarget.kt
package com.example.mmarecomp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class NutritionTarget(
    val id: String,
    @SerialName("user_id") val userId: String,
    val date: String,
    @SerialName("type_jour") val typeJour: TypeJour,
    @SerialName("calories_cible") val caloriesCible: Int,
    @SerialName("proteines_cible_g") val proteinesCibleG: Double,
    /** Nullables : l'historique antérieur à la périodisation glucidique n'en
     *  a pas, et une valeur inventée fausserait les récapitulatifs. */
    @SerialName("glucides_cible_g") val glucidesCibleG: Double? = null,
    @SerialName("lipides_cible_g") val lipidesCibleG: Double? = null,
    @SerialName("created_at") val createdAt: String? = null,
)

@Serializable
data class NewNutritionTarget(
    val date: String,
    @SerialName("type_jour") val typeJour: TypeJour,
    @SerialName("calories_cible") val caloriesCible: Int,
    @SerialName("proteines_cible_g") val proteinesCibleG: Double,
    @SerialName("glucides_cible_g") val glucidesCibleG: Double? = null,
    @SerialName("lipides_cible_g") val lipidesCibleG: Double? = null,
)
